aiohttp/connector.py

```diff
@@ -964,6 +975,15 @@ async def _wrap_create_connection(
             async with ceil_timeout(
                 timeout.sock_connect, ceil_threshold=timeout.ceil_threshold
             ):
+                if self._network_interface:
+                    *args, sock_hostname, sock_port = args
```

Comment:
That's a little weird. mypy seems to think `args` on the right-hand side is `List`, but `args` on the left-hand side is `Tuple`...

One solution is to assign to a new variable name, but looks like that'd make the code more complex. I'm fine with a type ignore here. Maybe we should file a mypy bug too (or check if one already exists).

Reply:
Actually, `*args` creates a list here, while `*args` as a parameter creates a tuple. That's why it's complaining.

Reply:
Ohh.. and how should it be mitigated?